packages/platform-ios/src/commands/runIOS/index.ts

```diff
@@ -368,10 +368,11 @@ function bootSimulator(selectedSimulator: Device) {
 }
 
 function getTargetBuildDir(buildSettings: string) {
-  const targetBuildMatch = /TARGET_BUILD_DIR = (.+)$/m.exec(buildSettings);
-  return targetBuildMatch && targetBuildMatch[1]
-    ? targetBuildMatch[1].trim()
-    : null;
+  const settings = JSON.parse(buildSettings);
+
+  return settings.length > 0
+    ? settings[1].buildSettings.TARGET_BUILD_DIR
+    : settings[0].buildSettings.TARGET_BUILD_DIR;
```

Comment:
if `settings.length` is 0 here, there's no `settings[0]`

Reply:
xcode will always return an array - in worst case scenario empty array. Probably it would make more sense to check other parts like `buildSettings` existence.

Reply:
@thymikee check update pls

Reply:
We discussed a good solution here, do you think you could do that instead? https://github.com/react-native-community/cli/pull/908#discussion_r381188106. Basically you can use settings.find to get the TARGET_BUILD_DIR that contains the app name.